NETProvider/src/FirebirdSql.Data.FirebirdClient/FirebirdClient/FbParameterCollection.cs

```diff
@@ -90,6 +91,21 @@ public override object SyncRoot
 			get { return ((ICollection)_parameters).SyncRoot; }
 		}
 
+		internal bool? _collectionHasParameterWithUnicodeName;
```

Comment:
> I don't want to recompute field on each change. I intentionally made computing the field only on-demand if IndexOf is called.

That was just an example. You may as well use two properties, one with value and one saying whether the value needs recomputing or not. What I'm saying is that the internal representation should not matter. Only the desired outcome of behavior.

I don't mind if you don't like it. I'll merge it anyway and then I'll change it. So it may as well stay as is.

> I think I can make the property private, but the field still need to be internal for tests

That's even worse.